### ISSUE TITLE
Layout/RescueEnsureAlignment incorrect behaviour when used as assignment

### ISSUE THREAD

**Issue by rhys-vdw** [rubocop-hq/rubocop#6918]
`Layout/RescueEnsureAlignment` does not support assignment

--------

## Expected behavior

Should consider indentation difference from first non-whitespace character in line containing `begin`.

```rb
values = begin
  get_values_array
rescue
  []
end
```

## Actual behavior

(for actual use case, example is simplified)

```
app/models/concerns/demographics.rb:21:5: C: Layout/RescueEnsureAlignment: rescue at 21, 4 is not aligned with begin at 19, 13.
    rescue ActiveRecord::StatementInvalid => error
    ^^^^^^
```

## Steps to reproduce the problem

Follow example.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
$ bundle exec rubocop -V
0.64.0 (using Parser 2.6.0.0, running on ruby 2.4.5 x86_64-darwin18)
```

**Comment by rhys-vdw** [comment]
Should probably behave in the same way as [`Layout/EndAlignment`](https://docs.rubocop.org/en/latest/cops_layout/#layoutendalignment).

**Comment by rhys-vdw** [comment]
This bot makes no sense. Why are you closing legitimate issues? To create to appearance that the project is maintained?  If you have lots of bugs and nobody's fixing them, then you _should_ have a high issue count.

**Comment by kepstin** [comment]
The alignment that the cop accepts here is hilariously bad :/
```ruby
values = begin
  get_values_array
         rescue
           []
end
```

This seems related to #6774 and #6254 and appears to have been introduced in #6246 (the old behaviour of having `rescue`/`ensure` match `end` gave good results here)

**Comment by jakeonrails** [comment]
I have a similar example of bad output but not from assignment:


```ruby
  coerce_input(lambda do |value, _ctx|
    URI(value)
  rescue URI::InvalidURIError
    nil
  end)

# Was autocorrected to:

  coerce_input(lambda do |value, _ctx|
    URI(value)
               rescue URI::InvalidURIError
                 nil
  end)
```

**Comment by Draiken** [comment]
I'm still seeing this. I agree with @rhys-vdw that it should have an option similar to the `Layout/EndAlignment`

**Comment by dvandersluis** [comment]
Agree with the previous commentators, this is burning us too, even in 0.75.1.

**Comment by ekalinichev** [comment]
FYI issue is still there

**Comment by mvz** [comment]
There's now Layout/BeginEndAlignment which, when enabled, should make the examples given here be flagged and/or corrected correctly.

**Comment by ianfixes** [comment]
via https://github.com/rubocop-hq/rubocop/pull/7531#issuecomment-602966770

I'm curious if this fix extends to cases with `memoize`, where in rubocop `0.77.0` it prefers this:

```ruby
  private memoize def do_something
    "Something"
                  rescue NameError
                    nil
  end
```

but without `memoize`, Rubocop acts sensibly and allows this:

```ruby
  private def do_something
    "Something"
  rescue NameError
    nil
  end
```

**Comment by jonas054** [comment]
The originally reported expected behavior is now the actual behavior with default configuration. Closing this issue.
